src/main/java/com/vlkan/pubsub/PubsubPublisher.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright 2019 Volkan Yazıcı
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permits and
+ * limitations under the License.
+ */
+
+package com.vlkan.pubsub;
+
+import com.vlkan.pubsub.model.PubsubPublishRequest;
+import com.vlkan.pubsub.model.PubsubPublishResponse;
+import reactor.core.publisher.Mono;
+
+import java.util.Objects;
+
+public class PubsubPublisher {
+
+    private final PubsubPublisherConfig config;
+
+    private final PubsubClient client;
+
+    private PubsubPublisher(Builder builder) {
+        this.config = builder.config;
+        this.client = builder.client;
+    }
+
+    public PubsubPublisherConfig getConfig() {
+        return config;
+    }
+
+    public PubsubClient getClient() {
+        return client;
+    }
+
+    public Mono<PubsubPublishResponse> publish(PubsubPublishRequest publishRequest) {
```

Comment:
Have you considered having a streaming mode, where `publish` accepts `Flux<PubsubPublishRequest>`? To reduce the overhead of a stream creation/subscription when one needs to send multiple requests, or stream from some source into pubsub?

Reply:
I will kindly pass this tip. `flatMap`ing a `Flux` has always been a two sided sword experience for me. Particularly, I find missing `concurrency` parameter a common source for unexpected load explosion. Yes, I can create three distinct methods:

1. `Mono<PubsubPublishResponse> publishOne(PubsubPublishRequest request)`
2. `Flux<PubsubPublishResponse> publishMany(Flux<PubsubPublishRequest> requests)`
3. `Flux<PubsubPublishResponse> publishMany(Flux<PubsubPublishRequest> requests, int concurrency)`

But I prefer to leave that up to the user.

Reply:
I was just concerned that if you leave it to the user then it may not be as optimal as if you would control it. And, since we're talking about RS and `publish` method, I guess the concurrency there would always be "just publish them ASAP" :D